modules/openid_auth/app/controllers/openid_auth/v2/validation_controller.rb

```diff
@@ -136,12 +131,25 @@ def should_validate_with_charon?(aud)
       end
 
       def validation_from_charon(duz, site)
-        response = RestClient.get(Settings.oidc.charon.endpoint,
-                                  { params: { duz: duz, site: site } })
-        response.code == 200
+        RestClient.get(Settings.oidc.charon.endpoint, { params: { duz: duz, site: site } })
+      rescue RestClient::ExceptionWithResponse => e
+        case e.response.code
+        when 400
+          json_response = JSON.parse(e.response.body)
+          raise error_klass(json_response['message'])
+        when 401
+          json_response = JSON.parse(e.response.body)
+          raise error_klass('Charon menu-code: ' + json_response['value'])
+        else
+          raise Common::Exceptions::TokenValidationError.new(
+            status: 500, code: 500, detail: 'Failed validation with Charon.'
+          )
+        end
       rescue => e
-        log_message_to_sentry('Failed validation with Charon', :error, body: e.message)
-        raise e
+        log_message_to_sentry('Error retrieving charon context for OIDC token: ' + e.message, :error)
```

Comment:
@beaugrantham-va, thoughts on logging unexpected charon responses to sentry?

This is similar to what we do fetching the launch context.

Reply:
Yes, this seems appropriate.